File: Python/Program/ProgramStudi.py
class ProgramStudi:
    
    def __init__(self, kode_prodi = "", nama_prodi = "", list_course = []):
        self.nama_prodi = nama_prodi
        self.kode_prodi = kode_prodi
        self.list_course = []
    
    def set_kode_prodi(self, kode_prodi):
        self.kode_prodi = kode_prodi

    def get_kode_prodi(self):
        return self.kode_prodi
    
    def set_nama_prodi(self, nama_prodi):
        self.nama_prodi = nama_prodi

    def get_nama_prodi(self):
        return self.nama_prodi

    def add_course(self, course):
        self.list_course.append(course)

    def show_course(self):
        print("| [Com] List Mata Kuliah")
        for course in self.list_course:
            print("| - ", course.nama_mata_kuliah)
    
    def display_data(self):
        print("| Kode Program Studi  : ", self.kode_prodi)
        print("| Nama Program Studi  : ", self.nama_prodi)
        self.show_course()
        print("+------------------------------------------+")

File: CPP/Program/ProgramStudi.cpp
using namespace std;
#include <bits/stdc++.h>
#include <vector>
#include "Course.cpp"

class ProgramStudi
{
public:
    string nama_prodi;
    string kode_prodi;
    vector<Course> courses;
    
    // Constructor
    ProgramStudi()
    {
        this->nama_prodi = "";
        this->kode_prodi = "";
    }

    ProgramStudi(string nama_prodi, string kode_prodi)
    {
        this->nama_prodi = nama_prodi;
        this->kode_prodi = kode_prodi;
    }

    // Setter & Getter

    void setNamaProdi(string nama_prodi)
    {
        this->nama_prodi = nama_prodi;
    }

    string getNamaProdi()
    {
        return nama_prodi;
    }

    void setKodeProdi(string kode_prodi)
    {
        this->kode_prodi = kode_prodi;
    }

    string getKodeProdi()
    {
        return kode_prodi;
    }

    void add_course(Course course) {
        courses.push_back(course);
    }

    // Destructor
    ~ProgramStudi() {}
};